src/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,556 @@
+"""
+module StateCheck
+
+# Imports from standard Julia packages
+using Formatting
+using MPI
+using Printf
+using StaticArrays
+using Statistics
+
+# Imports from ClimateMachine core
+import ClimateMachine.GenericCallbacks: EveryXSimulationSteps
+import ClimateMachine.MPIStateArrays: MPIStateArray
+import ClimateMachine.VariableTemplates: flattenednames
+
+####
+# For testing put a new function signature here!
+# Needs to go in src/Utilities/VariableTemplates/var_names.jl
+# This handles SMatrix case
+flattenednames(::Type{T}; prefix = "") where {T <: SArray} =
+    ntuple(i -> "$prefix[$i]", length(T))
+####
+
+struct vstat
+    max
+    min
+    mean
+    std
+end
+
+# Global functions to expose
+# sccreate - Create a state checker call back
+export sccreate
+export scdocheck
+export scprintref
+
+# ntFreqDef:: default frequency (in time steps) for output.
+# precDef  :: default precision used for formatted output table
+ntFreqDef = 10
+precDef = 15
+
+"""
+ sccreate :: Create a a "state check" call-back for one or more MPIStateArrays  \n
+             that will report basic statistics for the fields in the array.
+
+             Input:
+                fields: A required first argument that is an array of one or more
+                        MPIStateArrays and label string pair tuples.
+                        State array statistics will be reported for the named symbols
+                        in each MPIStateArray labeled with the label string.
+                ntFreq: An optional second argument with default value of 
+                        $ntFreqDef that sets how freuently (in time-step counts) the
+                        statistics are reported.
+             Named:
+                  prec: A named argument that sets number of decimal places to print for
+                        statistics, defaults to $precDef.
+
+             Return: 
+                sccb: A state checker that can be used in a callback().
+
+             Example:
+             julia> using ClimateMachine.VariableTemplates
+             julia> using StaticArrays
+             julia> using ClimateMachine.MPIStateArrays
+             julia> using MPI
+             julia> MPI.Init()
+             julia> T=Float64
+             julia> F1=@vars begin; ν∇u::SMatrix{3, 2, T, 6}; κ∇θ::SVector{3, T}; end
+             julia> F2=@vars begin; u::SVector{2, T}; θ::SVector{1, T}; end
+             julia> Q1=MPIStateArray{Float32,F1}(MPI.COMM_WORLD,ClimateMachine.array_type(),4,9,8);
+             julia> Q2=MPIStateArray{Float64,F2}(MPI.COMM_WORLD,ClimateMachine.array_type(),4,6,8);
+             julia> cb=StateCheck.sccreate([(Q1,"My gradients"),(Q2,"My fields")],1; prec=$precDef);
+             julia> cb()
+ ========================================================================================
+"""
+sccreate(
+    fields::Array{<:Tuple{<:MPIStateArray, String}},
+    ntFreq::Int = ntFreqDef;
+    prec = precDef,
+) = (
```

Comment:
inline function syntax should only be used for single-line functions. 
```suggestion
function sccreate(
    fields::Array{<:Tuple{<:MPIStateArray, String}},
    ntFreq::Int = ntFreqDef;
    prec = precDef,
)
```